pubsub/gcloud/aio/pubsub/metrics.py

```diff
@@ -0,0 +1,41 @@
+from gcloud.aio.auth import BUILD_GCLOUD_REST
+
+if BUILD_GCLOUD_REST:
+    pass
+else:
+    import prometheus_client
+
+    _NAMESPACE = 'gcloud_aio'
+    _SUBSYSTEM = 'pubsub'
+
+    CONSUME = prometheus_client.Counter(
+        'subscriber_consume',
+        'Counter of the outcomes of PubSub message consume attempts',
+        ['outcome'],
+        namespace=_NAMESPACE,
+        subsystem=_SUBSYSTEM,
+        unit='total')
+
+    CONSUME_LATENCY = prometheus_client.Histogram(
+        'subscriber_consume_latency',
+        'Histogram of PubSub message consume latencies',
+        ['phase'],
+        namespace=_NAMESPACE,
+        subsystem=_SUBSYSTEM,
+        unit='seconds')
+
+    BATCH_STATUS = prometheus_client.Counter(
+        'subscriber_batch_status',
+        'Counter for success/failure to process PubSub message batches',
```

Comment:
Related to my comment here https://github.com/talkiq/gcloud-aio/pull/425/files#r878185626, this doesn't say anything about being specific to the acker and nacker. You could say that it shouldn't be specific to only the acker and nacker components, but then something like summing over all of the components would not make sense if you had, for example, `producer` as a component to represent messages being pulled.